### PR TITLE
Automatically append arguments required by Electron to run on wayland

### PR DESCRIPTION
**Pull request by ppacher**
This should fix https://github.com/safing/portmaster-ui/issues/265 and is related to https://github.com/safing/portmaster/issues/495

### CODE REVIEW

File: cmds/portmaster-start/run.go [hunk 1]
```diff
@@ -121,6 +121,20 @@ func getExecArgs(opts *Options, cmdArgs []string) []string {
 	if stdinSignals {
 		args = append(args, "--input-signals")
 	}
+
+	// see https://www.freedesktop.org/software/systemd/man/pam_systemd.html#type=
+	if xdgSessionType := os.Getenv("XDG_SESSION_TYPE"); xdgSessionType == "wayland" && opts.Identifier == "app/portmaster-app.zip" {
```
Comment (dhaavi) [inline cmds/portmaster-start/run.go:126]: Not sure about the impact of `os.Getenv`, but we could pull the `opts.Identifier == "app/portmaster-app.zip"` check outside and only check the ENV if we are actually trying to launch the app.

File: cmds/portmaster-start/run.go [hunk 2]
```diff
@@ -121,6 +121,20 @@ func getExecArgs(opts *Options, cmdArgs []string) []string {
...
+		// we're running the Portmaster UI App under Wayland so make sure we add some arguments
+		// required by Electron
+		args = append(args,
+			[]string{
+				"--enable-features=UseOzonePlatform",
+				"--ozone-platform=wayland",
+			}...,
+		)
+
```
Comment (dhaavi) [inline cmds/portmaster-start/run.go:135]: No idea what's up here, but the linter in the editor should show.

File: cmds/portmaster-start/run.go [hunk 3]
```diff
@@ -121,6 +121,20 @@ func getExecArgs(opts *Options, cmdArgs []string) []string {
 	if stdinSignals {
 		args = append(args, "--input-signals")
 	}
+
+	// see https://www.freedesktop.org/software/systemd/man/pam_systemd.html#type=
```
Comment (dhaavi) [inline cmds/portmaster-start/run.go:125]: Does that mean this this only works when systemd is used?

**Review by dhaavi — CHANGES_REQUESTED** [review]

File: cmds/portmaster-start/run.go (hunk 3 above)
Comment (ppacher) [reply cmds/portmaster-start/run.go:125]: that environment variable is populated by systemd. I did not find any other reasonable way of detecting if wayland is used (and not just available or running for some other session). Most systems are using systemd anyway so this fix is for them. Non-systemd users can still workaround the issue by appending the args manually as suggested in  https://github.com/safing/portmaster/issues/495

File: cmds/portmaster-start/run.go (hunk 1 above)
Comment (ppacher) [reply cmds/portmaster-start/run.go:126]: environment variables are already part of the process memory and it's basically just a map lookup under a RLock/using a syscall so this check is negligible.

File: cmds/portmaster-start/run.go (hunk 3 above)
Comment (ppacher) [reply cmds/portmaster-start/run.go]: Though, for example GNOME also depends on that environment variable to decide in which mode to run so relying on that should be reasonable

**Review by dhaavi — APPROVED** [review]

**Pull request merged**